### DISCUSSION TITLE
Ruby on Rails Bundler Configuration

### DISCUSSION THREAD

**Discussion by henrikbjorn** [railwayapp/nixpacks#815] [Q&A]
Trying to wrap my head around Buildbacks, CNB and Nixpacks.

It seems that Nixpacks have a Ruby provider that works with rails, which will install bundler and the dependencies from Gemfile. However I can't seem to figure out how to configure bundler to do a deployment install.

Previously Bundler supported config through ENV variables like `BUNDLER_WITHOUT` etc. but that seems to have been deprecated.

Any guidance?

**Comment by henrikbjorn** [comment]
ref: https://github.com/railwayapp/nixpacks/pull/816

**Comment by WarnerAbediR** [comment]
Buildpacks, Cloud Native Buildpacks (CNB), and Nixpacks are all tools that aid in the development and deployment of software applications. Buildpacks are a tool for transforming application code into an executable artifact that can be deployed to a platform. CNBs are a newer version of buildpacks that have been designed to run in a more modular, cloud-native environment. Nixpacks are a packaging and deployment tool that uses the Nix package manager to create deterministic, reproducible builds.

Regarding configuring bundler for deployment install, it is true that the use of environment variables like BUNDLER_WITHOUT has been deprecated. Instead, Bundler now has a configuration file, "bundle config," that can be used to set deployment options like deployment mode, deployment path, and gem sources.

To configure Bundler for deployment install, you can create a new file called .bundle/config in your application's root directory. In this file, you can set the deployment mode to true by adding the line deployment: true. Additionally, you can set the deployment path by adding the line `pathpath: vendor/bundle.

Inbundle config command to.bundle/config file in
